catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -817,4 +778,41 @@ protected void validateGetResourceRequest(ResourceRequest getResourceRequest)
           "Cannot perform getResource with null attribute value or null attributeName, either passed in from endpoint, or as output from PreResourcePlugin");
     }
   }
+
+  private URI getDerivedUri(Metacard metacard, String qualifier) throws ResourceNotFoundException {
+    List<URI> urisWithMatchingQualifier = new ArrayList<>();
+    Attribute derivedResourcesAttribute = metacard.getAttribute(Core.DERIVED_RESOURCE_URI);
+
+    if (derivedResourcesAttribute != null) {
+      for (Serializable ser : derivedResourcesAttribute.getValues()) {
+        if (ser instanceof String) {
+          try {
+            URI uri = new URI((String) ser);
+
+            if (uri.getFragment().equals(qualifier)) {
+              urisWithMatchingQualifier.add(uri);
+            }
+          } catch (URISyntaxException e) {
+            LOGGER.error(
+                String.format(
+                    "Found derived URI without the proper syntax \"%s\" on metacard \"%s\"",
+                    ser, metacard.getId()));
+          }
+        }
+      }
+    }
+    if (urisWithMatchingQualifier.isEmpty()) {
+      throw new ResourceNotFoundException(
+          String.format(
+              "Couldn't find a derived URI with the qualifier \"%s\" on metacard \"%s\"",
+              qualifier, metacard.getId()));
+    }
+    if (urisWithMatchingQualifier.size() > 1) {
```

Comment:
Could you throw the exception immediately in the loop when you find multiple URIs with the qualifier? 
```java
  private URI getDerivedResourceUriForQualifier(Metacard metacard, String qualifier) throws ResourceNotFoundException {
    URI derivedResourceUriWithMatchingQualifier = null;
    final Attribute derivedResourceUriAttribute = metacard.getAttribute(Core.DERIVED_RESOURCE_URI);

    if (derivedResourceUriAttribute == null) {
      for (Serializable value : derivedResourceUriAttribute.getValues()) {
        if (value instanceof String) {
          try {
            URI uri = new URI((String) value);
            if (uri.getFragment().equals(qualifier)) {
              if (derivedResourceUriWithMatchingQualifier != null) {
                throw new ResourceNotFoundException(
                        String.format(
                                "Found multiple URIs with the qualifier \"%s\" on metacard \"%s\"",
                                qualifier, metacard.getId()));
              }
              derivedResourceUriWithMatchingQualifier = uri;
            }
          } catch (URISyntaxException e) {
            LOGGER.debug(
                    String.format(
                            "Found derived URI without the proper syntax \"%s\" on metacard \"%s\"",
                            value, metacard.getId()));
          }
        }
      }
    }

    if (derivedResourceUriWithMatchingQualifier == null) {
      throw new ResourceNotFoundException(
          String.format(
              "Could not find a derived URI with the qualifier \"%s\" on metacard \"%s\"",
              qualifier, metacard.getId()));
    }

    return derivedResourceUriWithMatchingQualifier;
  }
```